shimmingtoolbox/optimizer/lsq_optimizer.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import scipy.optimize as opt
+
+from shimmingtoolbox.optimizer.basic_optimizer import Optimizer
+
+
+class LSQ_Optimizer(Optimizer):
+
+    def _residuals(self, coef, unshimmed_vec, coil_mat):
+        """
+        Objective function to minimize
+        Args:
+            coef (numpy.ndarray): 1D array of channel coefficients
+            unshimmed_vec (numpy.ndarray): 1D flattened array (point) of the masked unshimmed map
+            coils_mat (numpy.ndarray): 2D flattened array (point, channel) of masked coils (axis 0 must align with unshimemd_vec)
+
+        Returns:
+            numpy.ndarray: Residuals for least squares optimization -- equivalent to flattened shimmed vector
+
+        """
+        self._error_if(unshimmed_vec.shape[0] != coil_mat.shape[0], f'Unshimmed ({unshimmed_vec.shape}) and coil ({coil_mat.shape}) arrays do not align on axis 0')
+        return unshimmed_vec + np.sum(coil_mat * coef, axis=1, keepdims=False)
+
+    def optimize(self, unshimmed, mask, mask_origin=(0, 0, 0), bounds=None):
+        """
+        Optimize unshimmed volume by varying current to each channel
+
+        Args:
+            unshimmed (numpy.ndarray): 3D B0 map
+            mask (numpy.ndarray): 3D integer mask used for the optimizer (only consider voxels with non-zero values).
+            mask_origin (tuple): Mask origin if mask volume does not cover unshimmed volume
+            bounds (list): List of ``(min, max)`` pairs for each coil channels. None
+               is used to specify no bound.
+
+        Returns:
+            numpy.ndarray: Coefficients corresponding to the coil profiles that minimize the objective function
+                           (coils.size)
+        """
+
+        # Check for sizing errors
+        self._check_sizing(unshimmed, mask, mask_origin=mask_origin, bounds=bounds)
+
+        m_x, m_y, m_z = mask_origin
+        m_X, m_Y, m_Z = mask.shape
```

Comment:
that would be snake_case (no capital allowed in variables as per [PEP8](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names))
```suggestion
        m_x, m_y, m_z = mask.shape
```

Reply:
I've rewritten this section -- my style was bad. Please check out the new section and let me know how it is.